keps/sig-network/2595-expanded-dns-config/README.md

```diff
@@ -0,0 +1,288 @@
+Rollback](#feature-enablement-and-rollback))
+- If the feature gate `ExpandedDNSConfig` is enabled, replace
+`MaxDNSSearchPaths` with `MaxDNSSearchPathsExpanded` and replace
+`MaxDNSSearchListChars` with `MaxDNSSearchListCharsExpanded` to allow expanded
+DNS configuration
+
+### Test Plan
+
+- Add unit tests of validating expanded DNS config
+
+### Graduation Criteria
+
+#### Alpha -> Beta Graduation
+
+- Address feedback from alpha
+- Sufficient testing
+
+#### Beta -> GA Graduation
+
+- Address feedback from beta
+- Sufficient number of users using the feature
+- Close on any remaining open issues & bugs
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+In clusters with older kubelets, old kubelets with `resolvConf` configured to
+exceed bounds throw warnings but do not fail. Eventually, old kubelets truncate
+the overage and apply the actual DNS resolver configuration.
+
+In clusters with a replicated control plane, all kube-apiservers should enable
+or disable the expanded DNS configuration feature.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+- **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: `ExpandedDNSConfig`
+    - Components depending on the feature gate:
+      - `kubelet`
+      - `kube-apiserver`
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+- **Does enabling the feature change any default behavior?**
+
+Enabling this feature allows kubernetes to have objects(Pod or PodTemplate) with
+the expanded DNS configuration.
+
+- **Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?**
+
+Yes, the feature can be disabled by disabling the feature gate.
+
+Once the feature is disabled, kube-apiserver will reject the pod having expanded
+DNS configuration and kubelet will create a resolver configuration excluding the
+overage.
+
+- **What happens if we reenable the feature if it was previously rolled back?**
+
+It should continue to work as expected.
+
+- **Are there any tests for feature enablement/disablement?**
+
+We will add unit tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+- **How can a rollout fail? Can it impact already running workloads?**
+
+N/A
+
+- **What specific metrics should inform a rollback?**
+
+N/A
+
+- **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+
+N/A
+
+- **Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?**
+
+N/A
+
+### Monitoring Requirements
+
+- **How can an operator determine if the feature is in use by workloads?**
+
+N/A
```

Comment:
They can query the api server and look for pods/templates/deployments with a larger-than-previously-allowed nameserver entry.